### ISSUE TITLE
Actions that I created find themselves in double in « Nautilus »...

### ISSUE THREAD

**Issue by Creteil** [bassmanitram/actions-for-nautilus#51]
I do not know why, but the specific actions that I created find themselves in double in  « Nautilus » :

![image](https://github.com/bassmanitram/actions-for-nautilus/assets/27310214/6df1e6da-79a0-4bdf-b0ed-14b1eda8bdff)

Do you know what can I do to fix it ?

**Comment by biokomiker** [comment]
Check whether you json file contains also double entries

Funnily I just opened an issue (for the wiki) about pdf compression. Please have a look - your solution might look similar. It took we a while to find a combination of paramenters which seemed to work fine for diverse pdfs

**Comment by Creteil** [comment]
This the 1st thing I have checked, and no, there is no double entries in my json file :

```
bigbob@bigbob-t480s:~$ cat ~/.local/share/actions-for-nautilus/config.json | jq | grep label
      "label": "Comparaison de 2 fichiers PDF's",
      "label": "Conversion de taille PDF's basse résolution (screen)",
      "label": "Conversion de taille PDF's moyenne résolution (ebook)",
      "label": "Conversion de taille PDF's haute résolution (printer)",
      "label": "Conversion de taille PDF's très haute résolution (prepress)",
      "label": "Conversion de fichier(s) mail (eml) en PDF(s)",
      "label": "Création d'une image ISO à partir d'un répertoire",
      "label": "Conversion résolution de fichiers Images vers 1024 points max",
bigbob@bigbob-t480s:~$
```

**Comment by Creteil** [comment]
@biokomiker, here is the commands behind my actions for nautilus :

```
bigbob@bigbob-t480s:~$ cat ~/.local/share/actions-for-nautilus/config.json | jq
{
  "actions": [
    {
      "type": "command",
      "label": "Comparaison de 2 fichiers PDF's",
      "command_line": "diffpdf %F",
      "min_items": 2,
      "max_items": 2,
      "permissions": "read",
      "mimetypes": [
        "application/pdf"
      ]
    },
    {
      "type": "command",
      "label": "Conversion de taille PDF's basse résolution (screen)",
      "command_line": "gs -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/screen -sOutputFile=%d/new-screen-%b %d/%b",
      "min_items": 1,
      "permissions": "read",
      "mimetypes": [
        "application/pdf"
      ]
    },
    {
      "type": "command",
      "label": "Conversion de taille PDF's moyenne résolution (ebook)",
      "command_line": "gs -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/ebook -sOutputFile=%d/new-ebook-%b %d/%b",
      "min_items": 1,
      "permissions": "read",
      "mimetypes": [
        "application/pdf"
      ]
    },
    {
      "type": "command",
      "label": "Conversion de taille PDF's haute résolution (printer)",
      "command_line": "gs -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/printer -sOutputFile=%d/new-printer-%b %d/%b",
      "min_items": 1,
      "permissions": "read",
      "mimetypes": [
        "application/pdf"
      ]
    },
    {
      "type": "command",
      "label": "Conversion de taille PDF's très haute résolution (prepress)",
      "command_line": "gs -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/prepress -sOutputFile=%d/new-prepress-%b %d/%b",
      "min_items": 1,
      "permissions": "read",
      "mimetypes": [
        "application/pdf"
      ]
    },
    {
      "type": "command",
      "label": "Conversion de fichier(s) mail (eml) en PDF(s)",
      "command_line": "java -jar /usr/local/bin/emailconverter.jar %d/%b",
      "min_items": 1,
      "permissions": "read",
      "mimetypes": [
        "message/rfc822"
      ]
    },
    {
      "type": "command",
      "label": "Création d'une image ISO à partir d'un répertoire",
      "command_line": "mkisofs -lJR -o %f.iso %f",
      "min_items": 1,
      "max_items": 1,
      "permissions": "read",
      "mimetypes": [
        "inode/directory"
      ]
    },
    {
      "type": "command",
      "label": "Conversion résolution de fichiers Images vers 1024 points max",
      "command_line": "convert -resize 1024x1024 %d/%b %d/max-1024-%b",
      "min_items": 1,
      "permissions": "read",
      "mimetypes": [
        "image/*"
      ]
    }
  ],
  "debug": false
}
bigbob@bigbob-t480s:~$
```

**Comment by bassmanitram** [comment]
Hey, @Creteil 

I just copied your config into mine and this is what I get... (well, apparently I have screwed up my Print key mapping, so) ... I get one entry per item...

What WOULD explain this is if, somehow, in the Nautilus extensions directory you have TWO versions of the extension. Nautilus would start each up and they would both look at the same config file - I just proved that that would happen. You _may_ have a local version (in `/$HOME/.local/share/nautilus-python/extensions`) and a "global" version installed with the `.deb` file?
